From 089666b33aa6debe1ba9baf8a38750d3a78883a1 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 27 Nov 2023 07:42:45 +0200
Subject: [PATCH] [FSW-334] header: rename preboot.h

As preboot has been removed from this project, the preboot.h header
that is widely used by all components of the project is renamed to
baremetal.h

Change-Id: I0be6d68b8229bb6ecb5cae2bc81739520674e33f
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/Makefile b/Makefile
index a1ea621..29ef5bc 100644
--- a/Makefile
+++ b/Makefile
@@ -84,7 +84,7 @@
 	@LANG=en_us date +'#define PREBOOT_DATE "%b %d %Y"' >> $@
 	@date +'#define PREBOOT_TIME "%T"' >> $@
 
-buildversion-habanalabs: include/preboot.h $(AUTO_GENERATED_FILE)
+buildversion-habanalabs: include/baremetal.h $(AUTO_GENERATED_FILE)
 	echo "VERSION_STR" | gcc -include $< -I./ -I ./lib/common/ -I ./$(HL_DEV_ARCH)/ -I$(SPECS_EMBEDDED_DIR)/$(SPECS_EMB_DEV) -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg -I$(SPECS_DIR)/ -I ./include/ -P -E -o - - | tail -1 | sed 's/\" \+\"//g;s/["\n]//g' | cut -d' ' -f2- | tr -d '\n' > $@
 
 $(PCI_PHY_FW_TARGET): $(HL_DEV_ARCH)/pcie_phy_fw.i
diff --git a/app/ext_rom/erom_pcie.c b/app/ext_rom/erom_pcie.c
index 3f965fa..3583b31 100644
--- a/app/ext_rom/erom_pcie.c
+++ b/app/ext_rom/erom_pcie.c
@@ -9,7 +9,7 @@
 #include <hl.h>
 #include <pll.h>
 #include <pci.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <security.h>
 #include <device.h>
 #include <spi_flash.h>
diff --git a/app/ext_rom/erom_recovery.c b/app/ext_rom/erom_recovery.c
index bdf8a86..c2e37fc 100644
--- a/app/ext_rom/erom_recovery.c
+++ b/app/ext_rom/erom_recovery.c
@@ -10,7 +10,7 @@
 #include <cmd_module.h>
 #include <stddef.h>
 #include <pll.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <security.h>
 #include <spi_flash.h>
 #include <device.h>
diff --git a/app/ext_rom/erom_tpm.c b/app/ext_rom/erom_tpm.c
index 9127e9d..3b6b129 100644
--- a/app/ext_rom/erom_tpm.c
+++ b/app/ext_rom/erom_tpm.c
@@ -6,7 +6,7 @@
  */
 
 #include <stdint.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <psoc_global_conf_regs.h>
 #include <psoc_security_regs.h>
 #include <hl_fw_if.h>
diff --git a/app/pcie_pif/pcie_pif.c b/app/pcie_pif/pcie_pif.c
index 085a5bd..30167dd 100644
--- a/app/pcie_pif/pcie_pif.c
+++ b/app/pcie_pif/pcie_pif.c
@@ -6,7 +6,7 @@
  */
 
 #include <stddef.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <device.h>
 #include <hl.h>
 #include <bitops.h>
diff --git a/app/pldm_bloader/pldm_bloader.c b/app/pldm_bloader/pldm_bloader.c
index e4273cc..18c44c9 100644
--- a/app/pldm_bloader/pldm_bloader.c
+++ b/app/pldm_bloader/pldm_bloader.c
@@ -6,7 +6,7 @@
  */
 
 #include <stddef.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <hl.h>
 #include <device.h>
 #include <uart.h>
diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 589ae34..bb83ae8 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -8,9 +8,9 @@
 #include <string.h>
 #include <hl_boot_if.h>
 #include <bitops.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <device.h>
-#include <hl_fw_if.h> /* requires preboot.h for __packed */
+#include <hl_fw_if.h> /* requires baremetal.h for __packed */
 
 #include "common.h"
 #include "uart.h"
diff --git a/arch/gaudi2/arc/asic.c b/arch/gaudi2/arc/asic.c
index e2e26b7..513df4b 100644
--- a/arch/gaudi2/arc/asic.c
+++ b/arch/gaudi2/arc/asic.c
@@ -19,7 +19,7 @@
 #include <cpu_if_masks.h>
 #include <psoc_reset_conf_regs.h>
 #include <bitops.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <uart.h>
 #include <pll.h>
 
diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 7260092..49cde90 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -29,7 +29,7 @@
 #include <common/pci_ids.h>
 #include <gaudi2_blocks_linux_driver.h>
 #include <arcboot_export.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <pll.h>
 #include <sha512.h>
 #include <uart.h>
diff --git a/arch/gaudi2/arc/onix.c b/arch/gaudi2/arc/onix.c
index 0bd31f1..f729889 100644
--- a/arch/gaudi2/arc/onix.c
+++ b/arch/gaudi2/arc/onix.c
@@ -5,7 +5,7 @@
  *
  */
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <efuse.h>
 #include <psoc_arc1_cfg_regs.h>
 #include <string.h>
diff --git a/arch/gaudi2/arc/pldm.c b/arch/gaudi2/arc/pldm.c
index 55a0b00..6bbc8a3 100644
--- a/arch/gaudi2/arc/pldm.c
+++ b/arch/gaudi2/arc/pldm.c
@@ -6,7 +6,7 @@
  */
 
 
-#include <preboot.h>
+#include <baremetal.h>
 
 #define SDRAM_BASE_FULL 0x1000FFFFFD000000ull
 
diff --git a/arch/gaudi2/arm/asic.c b/arch/gaudi2/arm/asic.c
index 7f916a0..2371531 100644
--- a/arch/gaudi2/arm/asic.c
+++ b/arch/gaudi2/arm/asic.c
@@ -7,7 +7,7 @@
 
 #include <string.h>
 #include <bitops.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <sha512.h>
 #include <boot_desc.h>
 
diff --git a/arch/gaudi2/arm/common.c b/arch/gaudi2/arm/common.c
index db2589b..14d316f 100644
--- a/arch/gaudi2/arm/common.c
+++ b/arch/gaudi2/arm/common.c
@@ -29,7 +29,7 @@
 
 #include <bitops.h>
 #include <sizes.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <pll.h>
 #include <uart.h>
 #include <string.h>
diff --git a/arch/gaudi2/arm/onix.c b/arch/gaudi2/arm/onix.c
index b0eee83..a71f528 100644
--- a/arch/gaudi2/arm/onix.c
+++ b/arch/gaudi2/arm/onix.c
@@ -6,7 +6,7 @@
  */
 
 
-#include <preboot.h>
+#include <baremetal.h>
 
 #define SYS_CLOCK_FREQUENCY	35000000ULL
 
diff --git a/arch/gaudi2/arm/pldm.c b/arch/gaudi2/arm/pldm.c
index 41e6d82..a1364f4 100644
--- a/arch/gaudi2/arm/pldm.c
+++ b/arch/gaudi2/arm/pldm.c
@@ -6,7 +6,7 @@
  */
 
 
-#include <preboot.h>
+#include <baremetal.h>
 
 #define SDRAM_BASE_FULL 0x1000FFFFFD000000ull
 
diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index aadedd7..9ca2387 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -15,7 +15,7 @@
 #include <cpu_mstr_if_axuser_regs.h>
 
 #include <string.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <pll.h>
 #include <uart.h>
 #include <pci.h>
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index f4e91b3..bc5ed29 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -30,7 +30,7 @@
 #include <cpu_ca53_cfg_regs.h>
 #include <cpu_if_regs.h>
 #include <cpu_if_masks.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <cmd_module.h>
 #include <gaudi2/gaudi2_reg_map.h>
 #include <gaudi2/gaudi2_boot_info.h>
diff --git a/arch/gaudi2/common/common.h b/arch/gaudi2/common/common.h
index c53fc12..9c797f3 100644
--- a/arch/gaudi2/common/common.h
+++ b/arch/gaudi2/common/common.h
@@ -6,7 +6,7 @@
 #ifndef ARCH_GAUDI2_COMMON_COMMON_H_
 #define ARCH_GAUDI2_COMMON_COMMON_H_
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <sizes.h>
 #include <limits.h>
 #include <arc_farm_arc0_dccm0_regs.h>
diff --git a/arch/gaudi2/common/onix.c b/arch/gaudi2/common/onix.c
index de5f98a..7a557a7 100644
--- a/arch/gaudi2/common/onix.c
+++ b/arch/gaudi2/common/onix.c
@@ -5,7 +5,7 @@
  *
  */
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <psoc_global_conf_regs.h>
 #include <cmd_module.h>
 #include <uart.h>
diff --git a/arch/gaudi2/common/pldm.c b/arch/gaudi2/common/pldm.c
index 3fbc047..e2b3430 100644
--- a/arch/gaudi2/common/pldm.c
+++ b/arch/gaudi2/common/pldm.c
@@ -6,7 +6,7 @@
  */
 
 #include <gaudi2_blocks_linux_driver.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <psoc_global_conf_regs.h>
 #include <psoc_cpu_pll_ctrl_regs.h>
 #include <pll.h>
diff --git a/arch/gaudi2/common/pll.c b/arch/gaudi2/common/pll.c
index 80de124..4ce6c0b 100644
--- a/arch/gaudi2/common/pll.c
+++ b/arch/gaudi2/common/pll.c
@@ -8,7 +8,7 @@
 
 #include <stddef.h>
 #include <bitops.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <pll.h>
 #include <uart.h>
 #include "common.h"
diff --git a/arch/gaudi2/common/reset.c b/arch/gaudi2/common/reset.c
index 90df013..25109a8 100644
--- a/arch/gaudi2/common/reset.c
+++ b/arch/gaudi2/common/reset.c
@@ -8,7 +8,7 @@
 #include <psoc_reset_conf_regs.h>
 
 #include <bitops.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <pll.h>
 #include <uart.h>
 
diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 4e53c46..2ccf56a 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -5,7 +5,7 @@
  *
  */
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <uart.h>
 #include <memory.h>
 #include <gaudi2/gaudi2.h>
diff --git a/arch/gaudi2/common/sha.c b/arch/gaudi2/common/sha.c
index ee15314..45cb2ea 100644
--- a/arch/gaudi2/common/sha.c
+++ b/arch/gaudi2/common/sha.c
@@ -6,7 +6,7 @@
  */
 
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <uart.h>
 
 #include "hl.h"
diff --git a/arch/gaudi2/common/tpm2.c b/arch/gaudi2/common/tpm2.c
index 00b31e2..47af4d8 100644
--- a/arch/gaudi2/common/tpm2.c
+++ b/arch/gaudi2/common/tpm2.c
@@ -9,7 +9,7 @@
 #include <stdarg.h>
 #include <tpm2.h>
 #include <uart.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <types.h>
 #include <string.h>
 #include <pll.h>
diff --git a/arch/gaudi3/arc/asic.c b/arch/gaudi3/arc/asic.c
index 44ca56d..5c5e3ad 100644
--- a/arch/gaudi3/arc/asic.c
+++ b/arch/gaudi3/arc/asic.c
@@ -14,7 +14,7 @@
 #include <common/pci_ids.h>
 
 #include <bitops.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <uart.h>
 #include <pll.h>
 #include <hl.h>
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 86936c2..3fcc759 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -48,7 +48,7 @@
 #include <sizes.h>
 
 #include "common.h"
-#include <preboot.h>
+#include <baremetal.h>
 #include "hl.h"
 #include <device.h>
 #include <boot_if.h>
diff --git a/arch/gaudi3/arc/erom_pcie.c b/arch/gaudi3/arc/erom_pcie.c
index 670a784..6846e2f 100644
--- a/arch/gaudi3/arc/erom_pcie.c
+++ b/arch/gaudi3/arc/erom_pcie.c
@@ -5,7 +5,7 @@
  *
  */
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <psoc_security_regs.h>
 #include <gaudi3_reg_map.h>
 #include <gaudi3_blocks_linux_driver.h>
diff --git a/arch/gaudi3/arc/erom_recovery.c b/arch/gaudi3/arc/erom_recovery.c
index c3056c5..24069cc 100644
--- a/arch/gaudi3/arc/erom_recovery.c
+++ b/arch/gaudi3/arc/erom_recovery.c
@@ -5,7 +5,7 @@
  *
  */
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <psoc_security_regs.h>
 #include <gaudi3_reg_map.h>
 #include <gaudi3_blocks_linux_driver.h>
diff --git a/arch/gaudi3/arc/erom_tpm.c b/arch/gaudi3/arc/erom_tpm.c
index bbda960..bcef2e6 100644
--- a/arch/gaudi3/arc/erom_tpm.c
+++ b/arch/gaudi3/arc/erom_tpm.c
@@ -5,7 +5,7 @@
  *
  */
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <psoc_security_regs.h>
 #include <gaudi3_reg_map.h>
 #include <gaudi3_blocks_linux_driver.h>
diff --git a/arch/gaudi3/arc/hl.h b/arch/gaudi3/arc/hl.h
index 1a714f5..63c02e4 100644
--- a/arch/gaudi3/arc/hl.h
+++ b/arch/gaudi3/arc/hl.h
@@ -7,7 +7,7 @@
 #define ARCH_GAUDI3_ARC_HL_H_
 
 #include <iopoll.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <parc_global_conf_regs.h>
 #include <parc_global_conf_masks.h>
 #include <gaudi3_fw_if.h>
diff --git a/arch/gaudi3/arc/onix.c b/arch/gaudi3/arc/onix.c
index 69af77d..02da182 100644
--- a/arch/gaudi3/arc/onix.c
+++ b/arch/gaudi3/arc/onix.c
@@ -10,7 +10,7 @@
 #include <common/pci_ids.h>
 
 #include <gaudi3/address_space.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <uart.h>
 
 #define SYS_CLOCK_FREQUENCY	35000000ULL
diff --git a/arch/gaudi3/arc/pldm.c b/arch/gaudi3/arc/pldm.c
index 3eb3497..5c7f369 100644
--- a/arch/gaudi3/arc/pldm.c
+++ b/arch/gaudi3/arc/pldm.c
@@ -10,7 +10,7 @@
 #include <common/pci_ids.h>
 #include <stddef.h>
 #include <bitops.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <uart.h>
 #include <pll.h>
 #include <arcboot_export_dev.h>
diff --git a/arch/gaudi3/common/common.c b/arch/gaudi3/common/common.c
index 5625a51..bc68d26 100644
--- a/arch/gaudi3/common/common.c
+++ b/arch/gaudi3/common/common.c
@@ -3,7 +3,7 @@
  * Copyright (C) 2022 HabanaLabs Ltd.
  */
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <hl.h>
 #include <gaudi3_blocks_linux_driver.h>
 #include <gaudi3_reg_map.h>
diff --git a/arch/gaudi3/common/pll.c b/arch/gaudi3/common/pll.c
index 979e955..456c183 100644
--- a/arch/gaudi3/common/pll.c
+++ b/arch/gaudi3/common/pll.c
@@ -6,7 +6,7 @@
 
 #include <stddef.h>
 #include <bitops.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <pll.h>
 #include "common.h"
 
diff --git a/arch/gaudi3/common/sha.c b/arch/gaudi3/common/sha.c
index 80dfed8..75c6772 100644
--- a/arch/gaudi3/common/sha.c
+++ b/arch/gaudi3/common/sha.c
@@ -6,7 +6,7 @@
  */
 
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <uart.h>
 
 #include "hl.h"
diff --git a/include/baremetal.h b/include/baremetal.h
new file mode 100644
index 0000000..741ee11
--- /dev/null
+++ b/include/baremetal.h
@@ -0,0 +1,288 @@
+/*
+ * Copyright (C) 2017 HabanaLabs Ltd.
+ *   Author: Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#ifndef COMMON_PREBOOT_H_
+#define COMMON_PREBOOT_H_
+
+#include <common.h>
+#include <sizes.h>
+
+#define ARRAY_SIZE(x)		(sizeof(x) / sizeof((x)[0]))
+#define ROUND_UP4(n)		(((n) + sizeof(u32) - 1) & ~(sizeof(u32) - 1))
+
+#ifdef __arc__
+#define wfe()	__asm__ __volatile__("wevt\n\t" : : )
+
+#define ARC_V2_IDENTITY_REG	0x004
+
+/*
+ * ARC1 released from reset first, so its ID in HW is always default 0.
+ * other ARCs ID is set relatively to ARC1.
+ * return normalized cpu id: arc0 = 0, arc1 = 1, arc2 = 2
+ */
+#define HL_ARC_V2_CORE_ID()					\
+({								\
+	s8 __ret;						\
+	__asm__ __volatile__("lr %0, [%1]\n"			\
+			     "xbfu %0, %0, 0xe8\n"		\
+			     : "=r"(__ret)			\
+			     : "i"(ARC_V2_IDENTITY_REG));	\
+	++__ret;						\
+})
+#endif
+
+#ifdef __aarch64__
+#define wfe()	__asm__ __volatile__ ("wfe\n\t" : : )
+#endif
+
+#define HL_UART_BAUDRATE	115200
+#define HL_SPI_FLASH_MAX_FREQ	25000000
+
+#define FIT_IMG_SPSRAM_MAX_SIZE		SZ_8K
+#define FIT_IMG_SPSRAM_END		(HL_SP_SRAM_BASE + HL_SP_SRAM_SIZE)
+#define FIT_IMG_SPSRAM_BASE		(FIT_IMG_SPSRAM_END - \
+					 FIT_IMG_SPSRAM_MAX_SIZE)
+
+
+#ifndef __ASSEMBLY__
+
+#include <stddef.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <types.h>
+#include "auto_generated.h"
+
+#define VERSION_MAX_LEN		128
+
+#define VERSION_STR	"version " PREBOOT_VER \
+	" (" PREBOOT_DATE " - " PREBOOT_TIME ")"
+
+#ifndef __packed
+#define __packed	__attribute__((packed))
+#endif
+#ifndef __weak
+#define __weak		__attribute__((weak))
+#endif
+#ifndef __aligned
+#define __aligned(x)	__attribute__((aligned(x)))
+#endif
+#ifndef __unused
+#define __unused	__attribute__((__unused__))
+#endif
+
+enum hl_debug_hw_func {
+	HL_DEBUG_HW_PCIE,
+	HL_DEBUG_HW_DRAM,
+	HL_DEBUG_HW_I2C,
+	HL_DEBUG_HW_DBG_CARD,
+};
+
+enum preboot_vers {
+	PREBOOT_BTL_VER,
+	PREBOOT_EFUSE_VER,
+};
+
+enum fit_sec_failure {
+	FIT_SEC_NONE = 0,
+	FIT_SEC_SVN_INVALID,
+	FIT_SEC_KEY_REVOKED,
+	FIT_SEC_SIGN_FAILED,
+};
+
+struct hl_nic_en {
+	bool is_valid;
+	uint32_t bitmask;
+};
+
+extern uintptr_t _start;	/* Start of ppboot/preboot region */
+extern uintptr_t __stack_top;	/* End of ppboot/preboot region */
+extern u32 __stack;		/* stack last cell value */
+extern struct hl_i2ci_device_status op_status; /* Contains OAM status info */
+
+#define STACK_CANARY_MAGIC_WORD	0x8BADF00D
+
+extern struct hl_fw_info_compat fw_info;
+/*
+ * TODO: Move to platform specific file once
+ * device id size is defined for other platforms
+ */
+#define FUSE_DEVID_SIZE	8
+
+extern u8 hl_asic_sn[FUSE_DEVID_SIZE];
+
+static inline u8 regmap_readb(u32 offset)
+{
+	u32 reg = *((volatile u32 *)(HL_CFG_BASE + offset));
+
+	return (u8)(reg & 0xFF);
+}
+
+static inline void regmap_writeb(u8 val, u32 offset)
+{
+	*((volatile u32 *)(HL_CFG_BASE + offset)) = (u32)val;
+}
+
+static inline u32 regmap_readl(u32 offset)
+{
+	u32 reg = *((volatile u32 *)(HL_CFG_BASE + offset));
+
+	return reg;
+}
+
+static inline void regmap_writel(u32 val, u32 offset)
+{
+	*((volatile u32 *)(HL_CFG_BASE + offset)) = val;
+}
+
+static inline void regmap_rmwritel(u32 addr, u32 data, u32 mask)
+{
+	u32 val;
+
+	val = regmap_readl(addr);
+	val &= ~mask;
+	val |= (data << (__builtin_ffsll(mask) - 1));
+
+	regmap_writel(val, addr);
+}
+
+u32 hl_skip_pcie_init(void);
+u32 hl_skip_fw_cfg(void);
+void hl_uart_init(void);
+u32 hl_spi_baudr_get(void);
+void hl_plat_early_init(void);
+void hl_ts_init(void);
+void hl_ts_shutdown(void);
+void hl_plat_init(void);
+bool hl_is_debug_mode(enum hl_debug_hw_func func);
+bool hl_dev_spec_debug_mode(enum hl_debug_hw_func func);
+bool hl_is_jtag_enabled(void);
+void hl_enter_recovery(void);
+void hl_dev_reset(void);
+void hl_esc_key_handle(void);
+u32 hl_rst_src_get(void);
+void hl_reset_src_init(void);
+void hl_set_reset_cause(uint32_t reset_cause);
+
+void preboot_pcie_wait_for_link(void);
+void preboot_delay(void);
+void preboot_putc_virt_uart(const char c);
+void preboot_sync_btl(void);
+void preboot_led_on(void);
+void preboot_pcie_enable(void);
+void preboot_pcie_exec_hw_wa(void);
+void preboot_nic_en(u32 nic_en_);
+bool preboot_host_gpio_asserted(void);
+void preboot_sp_sram_init(void);
+void preboot_cond_arc0_reset(void);
+
+void ppboot_cp_preboot(u64 loadaddr, u32 offset, u32 size, char *name);
+void image_start(u64 loadaddr);
+u32 sp_sram_readl(u32 offset);
+void hl_sram_self_repair(void);
+
+struct preboot_fuse_ver_data {
+	char *table_ver;
+	size_t table_ver_sz;
+	char *dev_id;
+	size_t dev_id_sz;
+	char *fab;
+	size_t fab_sz;
+	char *lot;
+	size_t lot_sz;
+	char *wafer;
+	size_t wafer_sz;
+	char *xaddr;
+	size_t xaddr_sz;
+	char *yaddr;
+	size_t yaddr_sz;
+};
+
+void preboot_fuse_version_serialize(char *ver, size_t size,
+				    struct preboot_fuse_ver_data *fuse_data,
+				    bool is_reverse);
+bool preboot_sram_is_valid_address_range(u64 addr, int len);
+bool preboot_is_valid_address_range(u64 addr, int len);
+uintptr_t preboot_get_elbi_reg_base(void);
+u32 preboot_get_elbi_reg_size(void);
+void preboot_dynamic_img_elbi_reg_cleanup(void *map_addr, u32 size);
+bool preboot_is_overlap_address_range(u64 addr, int len);
+
+bool preboot_is_bis_compliant_mode(void);
+enum hl_fw_en_caps;
+
+void preboot_dev_enable_caps(void);
+void preboot_set_enabled_cap(enum hl_fw_en_caps en_cap);
+void preboot_finalize_enabled_caps(void);
+void preboot_ack_irq(void);
+const char *preboot_get_version(enum preboot_vers ver);
+void hl_ram_init(void);
+void hl_pci_iatu_init(void);
+bool hl_pci_link_is_up(void);
+void hl_pcie_enable_access(void);
+void hl_pcie_disable_access(void);
+void hl_pcie_enable_mem_access(void);
+void hl_pcie_disable_mem_access(void);
+
+void preboot_finalize_fw_info(void);
+void preboot_finalize_nic_en(void);
+void preboot_boot_info_set_fw_info(void);
+void preboot_finalize_rma_tables(void);
+void preboot_finalize_fw_upd_closure(void);
+
+bool preboot_is_image_sig_supported(void);
+int preboot_verify_img_sig(const void *fit, size_t img_size,
+			   enum fit_sec_failure *res, u32 *key_idx);
+size_t preboot_get_img_sec_desc_size(void);
+void preboot_start_image(u64 startaddr, bool valid_arc_img);
+
+u32 preboot_error_read_and_clr(u32 *major, u32 *minor);
+bool preboot_error_inform(u32 major, u32 minor);
+bool preboot_error_is_pending(void);
+
+void preboot_hbm_credits_init(void);
+void preboot_rate_limiter_init(void);
+void preboot_e2e_credits_init(void);
+
+int preboot_set_sram_sf_access(bool is_enable);
+void preboot_device_id_init(void);
+u32 preboot_device_id_get(void);
+bool is_pcie_initialized(void);
+void preboot_init_pcie_phy_fw_update(void);
+void preboot_disable_pcie_ltr(void);
+void preboot_init_pcie_phy(void);
+void preboot_engines_init(void);
+void preboot_clear_wd_gpio(void);
+void preboot_set_eeprom_defaults_en(u8 eeprom_defaults_on_err);
+void preboot_finalize_uboot_override(void);
+void preboot_update_bar_sizes(void);
+
+void ppboot_init_stack_mon_dev_specific(void);
+void ppboot_check_stack_mon_dev_specific(void);
+void ppboot_pcie_drain_init(void);
+void ppboot_security_init(void);
+void hl_dev_security_init(void);
+int ppboot_boot_info_get(u32 type, void *data, u32 max_size, u32 *copy_size);
+void ppboot_plls_init(void);
+
+void preboot_main(void);
+void preboot_version(void);
+void preboot_dev_spec_legacy_vers_setup(enum preboot_vers vers_type, u32 offset);
+void ppboot_secure_arc(void);
+
+void hl_pif_report_err(u32 err_bit);
+void hl_pif_report_sts(u32 status_bit);
+void hl_pif_clear_err(u32 err_bit);
+void hl_pif_clear_sts(u32 status_bit);
+void preboot_security_sp_sram_init(void);
+u32 hl_get_pll_div_fix(void);
+void preboot_boot_info_init(void);
+void ppboot_boot_info_init(void);
+void preboot_recovery_flag(bool en);
+
+#endif /* __ASSEMBLY__ */
+
+#endif /* COMMON_PREBOOT_H_ */
diff --git a/include/boot_desc.h b/include/boot_desc.h
index cd2defa..0e38a29 100644
--- a/include/boot_desc.h
+++ b/include/boot_desc.h
@@ -7,7 +7,7 @@
 #ifndef _BOOT_DESC_H_
 #define _BOOT_DESC_H_
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <hl_fw_if.h>
 #include <hl_boot_desc.h>
 #include <sha.h>
diff --git a/include/boot_info.h b/include/boot_info.h
index 9917ae1..98f78db 100644
--- a/include/boot_info.h
+++ b/include/boot_info.h
@@ -7,7 +7,7 @@
 #ifndef _BOOT_INFO_H_
 #define _BOOT_INFO_H_
 
-#include <preboot.h>
+#include <baremetal.h>
 
 #define BOOT_INFO_SUCCESS	0
 #define BOOT_INFO_MEMERR	1
diff --git a/include/iopoll.h b/include/iopoll.h
index f4e6264..e9136e9 100644
--- a/include/iopoll.h
+++ b/include/iopoll.h
@@ -6,7 +6,7 @@
 #ifndef _LINUX_IOPOLL_H
 #define _LINUX_IOPOLL_H
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <pll.h>
 #include <errno.h>
 
diff --git a/include/preboot.h b/include/preboot.h
deleted file mode 100644
index 741ee11..0000000
--- a/include/preboot.h
+++ /dev/null
@@ -1,288 +0,0 @@
-/*
- * Copyright (C) 2017 HabanaLabs Ltd.
- *   Author: Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
-#ifndef COMMON_PREBOOT_H_
-#define COMMON_PREBOOT_H_
-
-#include <common.h>
-#include <sizes.h>
-
-#define ARRAY_SIZE(x)		(sizeof(x) / sizeof((x)[0]))
-#define ROUND_UP4(n)		(((n) + sizeof(u32) - 1) & ~(sizeof(u32) - 1))
-
-#ifdef __arc__
-#define wfe()	__asm__ __volatile__("wevt\n\t" : : )
-
-#define ARC_V2_IDENTITY_REG	0x004
-
-/*
- * ARC1 released from reset first, so its ID in HW is always default 0.
- * other ARCs ID is set relatively to ARC1.
- * return normalized cpu id: arc0 = 0, arc1 = 1, arc2 = 2
- */
-#define HL_ARC_V2_CORE_ID()					\
-({								\
-	s8 __ret;						\
-	__asm__ __volatile__("lr %0, [%1]\n"			\
-			     "xbfu %0, %0, 0xe8\n"		\
-			     : "=r"(__ret)			\
-			     : "i"(ARC_V2_IDENTITY_REG));	\
-	++__ret;						\
-})
-#endif
-
-#ifdef __aarch64__
-#define wfe()	__asm__ __volatile__ ("wfe\n\t" : : )
-#endif
-
-#define HL_UART_BAUDRATE	115200
-#define HL_SPI_FLASH_MAX_FREQ	25000000
-
-#define FIT_IMG_SPSRAM_MAX_SIZE		SZ_8K
-#define FIT_IMG_SPSRAM_END		(HL_SP_SRAM_BASE + HL_SP_SRAM_SIZE)
-#define FIT_IMG_SPSRAM_BASE		(FIT_IMG_SPSRAM_END - \
-					 FIT_IMG_SPSRAM_MAX_SIZE)
-
-
-#ifndef __ASSEMBLY__
-
-#include <stddef.h>
-#include <stdint.h>
-#include <stdbool.h>
-#include <types.h>
-#include "auto_generated.h"
-
-#define VERSION_MAX_LEN		128
-
-#define VERSION_STR	"version " PREBOOT_VER \
-	" (" PREBOOT_DATE " - " PREBOOT_TIME ")"
-
-#ifndef __packed
-#define __packed	__attribute__((packed))
-#endif
-#ifndef __weak
-#define __weak		__attribute__((weak))
-#endif
-#ifndef __aligned
-#define __aligned(x)	__attribute__((aligned(x)))
-#endif
-#ifndef __unused
-#define __unused	__attribute__((__unused__))
-#endif
-
-enum hl_debug_hw_func {
-	HL_DEBUG_HW_PCIE,
-	HL_DEBUG_HW_DRAM,
-	HL_DEBUG_HW_I2C,
-	HL_DEBUG_HW_DBG_CARD,
-};
-
-enum preboot_vers {
-	PREBOOT_BTL_VER,
-	PREBOOT_EFUSE_VER,
-};
-
-enum fit_sec_failure {
-	FIT_SEC_NONE = 0,
-	FIT_SEC_SVN_INVALID,
-	FIT_SEC_KEY_REVOKED,
-	FIT_SEC_SIGN_FAILED,
-};
-
-struct hl_nic_en {
-	bool is_valid;
-	uint32_t bitmask;
-};
-
-extern uintptr_t _start;	/* Start of ppboot/preboot region */
-extern uintptr_t __stack_top;	/* End of ppboot/preboot region */
-extern u32 __stack;		/* stack last cell value */
-extern struct hl_i2ci_device_status op_status; /* Contains OAM status info */
-
-#define STACK_CANARY_MAGIC_WORD	0x8BADF00D
-
-extern struct hl_fw_info_compat fw_info;
-/*
- * TODO: Move to platform specific file once
- * device id size is defined for other platforms
- */
-#define FUSE_DEVID_SIZE	8
-
-extern u8 hl_asic_sn[FUSE_DEVID_SIZE];
-
-static inline u8 regmap_readb(u32 offset)
-{
-	u32 reg = *((volatile u32 *)(HL_CFG_BASE + offset));
-
-	return (u8)(reg & 0xFF);
-}
-
-static inline void regmap_writeb(u8 val, u32 offset)
-{
-	*((volatile u32 *)(HL_CFG_BASE + offset)) = (u32)val;
-}
-
-static inline u32 regmap_readl(u32 offset)
-{
-	u32 reg = *((volatile u32 *)(HL_CFG_BASE + offset));
-
-	return reg;
-}
-
-static inline void regmap_writel(u32 val, u32 offset)
-{
-	*((volatile u32 *)(HL_CFG_BASE + offset)) = val;
-}
-
-static inline void regmap_rmwritel(u32 addr, u32 data, u32 mask)
-{
-	u32 val;
-
-	val = regmap_readl(addr);
-	val &= ~mask;
-	val |= (data << (__builtin_ffsll(mask) - 1));
-
-	regmap_writel(val, addr);
-}
-
-u32 hl_skip_pcie_init(void);
-u32 hl_skip_fw_cfg(void);
-void hl_uart_init(void);
-u32 hl_spi_baudr_get(void);
-void hl_plat_early_init(void);
-void hl_ts_init(void);
-void hl_ts_shutdown(void);
-void hl_plat_init(void);
-bool hl_is_debug_mode(enum hl_debug_hw_func func);
-bool hl_dev_spec_debug_mode(enum hl_debug_hw_func func);
-bool hl_is_jtag_enabled(void);
-void hl_enter_recovery(void);
-void hl_dev_reset(void);
-void hl_esc_key_handle(void);
-u32 hl_rst_src_get(void);
-void hl_reset_src_init(void);
-void hl_set_reset_cause(uint32_t reset_cause);
-
-void preboot_pcie_wait_for_link(void);
-void preboot_delay(void);
-void preboot_putc_virt_uart(const char c);
-void preboot_sync_btl(void);
-void preboot_led_on(void);
-void preboot_pcie_enable(void);
-void preboot_pcie_exec_hw_wa(void);
-void preboot_nic_en(u32 nic_en_);
-bool preboot_host_gpio_asserted(void);
-void preboot_sp_sram_init(void);
-void preboot_cond_arc0_reset(void);
-
-void ppboot_cp_preboot(u64 loadaddr, u32 offset, u32 size, char *name);
-void image_start(u64 loadaddr);
-u32 sp_sram_readl(u32 offset);
-void hl_sram_self_repair(void);
-
-struct preboot_fuse_ver_data {
-	char *table_ver;
-	size_t table_ver_sz;
-	char *dev_id;
-	size_t dev_id_sz;
-	char *fab;
-	size_t fab_sz;
-	char *lot;
-	size_t lot_sz;
-	char *wafer;
-	size_t wafer_sz;
-	char *xaddr;
-	size_t xaddr_sz;
-	char *yaddr;
-	size_t yaddr_sz;
-};
-
-void preboot_fuse_version_serialize(char *ver, size_t size,
-				    struct preboot_fuse_ver_data *fuse_data,
-				    bool is_reverse);
-bool preboot_sram_is_valid_address_range(u64 addr, int len);
-bool preboot_is_valid_address_range(u64 addr, int len);
-uintptr_t preboot_get_elbi_reg_base(void);
-u32 preboot_get_elbi_reg_size(void);
-void preboot_dynamic_img_elbi_reg_cleanup(void *map_addr, u32 size);
-bool preboot_is_overlap_address_range(u64 addr, int len);
-
-bool preboot_is_bis_compliant_mode(void);
-enum hl_fw_en_caps;
-
-void preboot_dev_enable_caps(void);
-void preboot_set_enabled_cap(enum hl_fw_en_caps en_cap);
-void preboot_finalize_enabled_caps(void);
-void preboot_ack_irq(void);
-const char *preboot_get_version(enum preboot_vers ver);
-void hl_ram_init(void);
-void hl_pci_iatu_init(void);
-bool hl_pci_link_is_up(void);
-void hl_pcie_enable_access(void);
-void hl_pcie_disable_access(void);
-void hl_pcie_enable_mem_access(void);
-void hl_pcie_disable_mem_access(void);
-
-void preboot_finalize_fw_info(void);
-void preboot_finalize_nic_en(void);
-void preboot_boot_info_set_fw_info(void);
-void preboot_finalize_rma_tables(void);
-void preboot_finalize_fw_upd_closure(void);
-
-bool preboot_is_image_sig_supported(void);
-int preboot_verify_img_sig(const void *fit, size_t img_size,
-			   enum fit_sec_failure *res, u32 *key_idx);
-size_t preboot_get_img_sec_desc_size(void);
-void preboot_start_image(u64 startaddr, bool valid_arc_img);
-
-u32 preboot_error_read_and_clr(u32 *major, u32 *minor);
-bool preboot_error_inform(u32 major, u32 minor);
-bool preboot_error_is_pending(void);
-
-void preboot_hbm_credits_init(void);
-void preboot_rate_limiter_init(void);
-void preboot_e2e_credits_init(void);
-
-int preboot_set_sram_sf_access(bool is_enable);
-void preboot_device_id_init(void);
-u32 preboot_device_id_get(void);
-bool is_pcie_initialized(void);
-void preboot_init_pcie_phy_fw_update(void);
-void preboot_disable_pcie_ltr(void);
-void preboot_init_pcie_phy(void);
-void preboot_engines_init(void);
-void preboot_clear_wd_gpio(void);
-void preboot_set_eeprom_defaults_en(u8 eeprom_defaults_on_err);
-void preboot_finalize_uboot_override(void);
-void preboot_update_bar_sizes(void);
-
-void ppboot_init_stack_mon_dev_specific(void);
-void ppboot_check_stack_mon_dev_specific(void);
-void ppboot_pcie_drain_init(void);
-void ppboot_security_init(void);
-void hl_dev_security_init(void);
-int ppboot_boot_info_get(u32 type, void *data, u32 max_size, u32 *copy_size);
-void ppboot_plls_init(void);
-
-void preboot_main(void);
-void preboot_version(void);
-void preboot_dev_spec_legacy_vers_setup(enum preboot_vers vers_type, u32 offset);
-void ppboot_secure_arc(void);
-
-void hl_pif_report_err(u32 err_bit);
-void hl_pif_report_sts(u32 status_bit);
-void hl_pif_clear_err(u32 err_bit);
-void hl_pif_clear_sts(u32 status_bit);
-void preboot_security_sp_sram_init(void);
-u32 hl_get_pll_div_fix(void);
-void preboot_boot_info_init(void);
-void ppboot_boot_info_init(void);
-void preboot_recovery_flag(bool en);
-
-#endif /* __ASSEMBLY__ */
-
-#endif /* COMMON_PREBOOT_H_ */
diff --git a/include/rma_info.h b/include/rma_info.h
index dad4267..8857c8c 100644
--- a/include/rma_info.h
+++ b/include/rma_info.h
@@ -7,7 +7,7 @@
 #ifndef _RMA_INFO_H_
 #define _RMA_INFO_H_
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <hl_i2c_indirect.h>
 #include <rma_if.h>
 
diff --git a/lib/common/boot_desc.c b/lib/common/boot_desc.c
index e4550ba..b7d40f9 100644
--- a/lib/common/boot_desc.c
+++ b/lib/common/boot_desc.c
@@ -8,7 +8,7 @@
 #include <string.h>
 #include <stdbool.h>
 #include <stdint.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <hl_fw_if.h>		/* requires stdint.h */
 #include <hl_boot_desc.h>
 #include <common.h>
diff --git a/lib/common/boot_if.c b/lib/common/boot_if.c
index 25ed55f..dc6f4b5 100644
--- a/lib/common/boot_if.c
+++ b/lib/common/boot_if.c
@@ -7,7 +7,7 @@
 #include <stdbool.h>
 #include <stdint.h>
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <device.h>
 #include <hl_boot_if.h>			/* requires stdint.h */
 #include <bitops.h>
diff --git a/lib/common/boot_info.c b/lib/common/boot_info.c
index 3a16c4e..3e7a59d 100644
--- a/lib/common/boot_info.c
+++ b/lib/common/boot_info.c
@@ -6,7 +6,7 @@
  */
 #include <string.h>
 #include <stdbool.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <hl_boot_desc.h>
 #include <common.h>
 #include <sha256.h>
diff --git a/lib/common/cmd_module.c b/lib/common/cmd_module.c
index 3190f40..12696fa 100644
--- a/lib/common/cmd_module.c
+++ b/lib/common/cmd_module.c
@@ -5,7 +5,7 @@
  *
  */
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <boot_if.h>
 
 #include "uart.h"
diff --git a/lib/common/crc32.c b/lib/common/crc32.c
index c999a1d..28e0a59 100644
--- a/lib/common/crc32.c
+++ b/lib/common/crc32.c
@@ -5,7 +5,7 @@
  *
  */
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <uart.h>
 
 /*
diff --git a/lib/common/efuse.c b/lib/common/efuse.c
index 6026725..b5721fb 100644
--- a/lib/common/efuse.c
+++ b/lib/common/efuse.c
@@ -6,12 +6,12 @@
  */
 
 #include <unistd.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <device.h>
 #include <boot_if.h>
 #include "efuse.h"
 #include "common.h"
-#include "preboot.h"
+#include "baremetal.h"
 #include "uart.h"
 #include "pll.h"
 
diff --git a/lib/common/efuse.h b/lib/common/efuse.h
index 2aa57a6..d32cabd 100644
--- a/lib/common/efuse.h
+++ b/lib/common/efuse.h
@@ -8,7 +8,7 @@
 #ifndef _EFUSE_H_
 #define _EFUSE_H_
 
-#include <preboot.h>
+#include <baremetal.h>
 
 #define EFUSE_READ_RANGE_ALL		0x3F80
 #define EFUSE_PRODUCT_ID_OFFSET		0x1
diff --git a/lib/common/lzo/lzo1x_decompress.c b/lib/common/lzo/lzo1x_decompress.c
index 9b7a1eb..acad4b5 100644
--- a/lib/common/lzo/lzo1x_decompress.c
+++ b/lib/common/lzo/lzo1x_decompress.c
@@ -11,7 +11,7 @@
  *  Richard Purdie <rpurdie@openedhand.com>
  */
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <string.h>
 
 #include "lzo.h"
diff --git a/lib/common/module_i2c_slave.c b/lib/common/module_i2c_slave.c
index 9b3f80d..cbda9cf 100644
--- a/lib/common/module_i2c_slave.c
+++ b/lib/common/module_i2c_slave.c
@@ -7,9 +7,9 @@
 
 #include <stdint.h>
 #include <stddef.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <device.h>
-#include <hl_fw_if.h> /* requires preboot.h for __packed */
+#include <hl_fw_if.h> /* requires baremetal.h for __packed */
 #include <preboot/thermal.h>
 #include <hl_i2c_slave_events.h>
 #include <crc8.h>
diff --git a/lib/common/module_pci.c b/lib/common/module_pci.c
index 0eede9e..dc96982 100644
--- a/lib/common/module_pci.c
+++ b/lib/common/module_pci.c
@@ -6,7 +6,7 @@
 
 
 #include <bitops.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <device.h>
 
 #include "cmd_module.h"
diff --git a/lib/common/pci.c b/lib/common/pci.c
index fa3c021..38c1618 100644
--- a/lib/common/pci.c
+++ b/lib/common/pci.c
@@ -7,7 +7,7 @@
 
 #include <common/pci_ids.h>
 #include <bitops.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <device.h>
 #include <sizes.h>
 #include "pci.h"
diff --git a/lib/common/pif_ts.c b/lib/common/pif_ts.c
index 6ca8238..3403413 100644
--- a/lib/common/pif_ts.c
+++ b/lib/common/pif_ts.c
@@ -4,7 +4,7 @@
  * All Rights Reserved.
  */
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <device.h>
 #include <bitops.h>
 #include <hl_pif_arc_if.h>
diff --git a/lib/common/pll.c b/lib/common/pll.c
index f741a52..0370013 100644
--- a/lib/common/pll.c
+++ b/lib/common/pll.c
@@ -5,7 +5,7 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-#include "preboot.h"
+#include "baremetal.h"
 #include "uart.h"
 #include "pll.h"
 #include "hl.h"
diff --git a/lib/common/spi.c b/lib/common/spi.c
index 8db05c1..59d69cb 100644
--- a/lib/common/spi.c
+++ b/lib/common/spi.c
@@ -8,7 +8,7 @@
 #include <stddef.h>
 
 #include <bitops.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include "common.h"
 #include <spi.h>
 #include "uart.h"
diff --git a/lib/common/spi_flash.c b/lib/common/spi_flash.c
index a55363e..dcf4941 100644
--- a/lib/common/spi_flash.c
+++ b/lib/common/spi_flash.c
@@ -9,7 +9,7 @@
 #include <stdbool.h>
 
 #include <sizes.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include "common.h"
 #include <spi.h>
 #include "uart.h"
diff --git a/lib/common/uart.c b/lib/common/uart.c
index 690117f..0d27aed 100644
--- a/lib/common/uart.c
+++ b/lib/common/uart.c
@@ -6,7 +6,7 @@
  */
 #include <stddef.h>
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <device.h>
 #include <sizes.h>
 #include "uart.h"
diff --git a/lib/preboot/crc8.c b/lib/preboot/crc8.c
index cfd9869..d2aab91 100644
--- a/lib/preboot/crc8.c
+++ b/lib/preboot/crc8.c
@@ -5,7 +5,7 @@
  *
  */
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <crc8.h>
 
 /* This code was copied from from linux/lib/crc8.c */
diff --git a/lib/preboot/fit.c b/lib/preboot/fit.c
index 7c4baa2..1d343c4 100644
--- a/lib/preboot/fit.c
+++ b/lib/preboot/fit.c
@@ -10,8 +10,8 @@
 #include <bitops.h>
 #include <security.h>
 #include <hl_boot_if.h>
-#include <preboot.h>
-#include <hl_boot_desc.h> /* requires preboot.h for __packed */
+#include <baremetal.h>
+#include <hl_boot_desc.h> /* requires baremetal.h for __packed */
 #include <boot_if.h>
 #include <fit_load_type.h>
 #include <boot_info.h>
diff --git a/lib/preboot/fit_rsa.h b/lib/preboot/fit_rsa.h
index 0efa956..be24bc1 100644
--- a/lib/preboot/fit_rsa.h
+++ b/lib/preboot/fit_rsa.h
@@ -6,7 +6,7 @@
 #ifndef COMMON_FIT_RSA_H_
 #define COMMON_FIT_RSA_H_
 
-#include <preboot.h>
+#include <baremetal.h>
 
 struct rsa_key_prop {
 	const void *rr;			/* R^2 can be treated as byte array */
diff --git a/lib/preboot/i2cslave_indirect.c b/lib/preboot/i2cslave_indirect.c
index 552c39d..84bdb85 100644
--- a/lib/preboot/i2cslave_indirect.c
+++ b/lib/preboot/i2cslave_indirect.c
@@ -8,7 +8,7 @@
 #include <stdbool.h>
 #include <string.h>
 #include <bitops.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <address_space.h>
 #include <uart.h>
 #include <common/module_i2c_slave.h>
diff --git a/lib/preboot/memory.c b/lib/preboot/memory.c
index b60acb4..414bbab 100644
--- a/lib/preboot/memory.c
+++ b/lib/preboot/memory.c
@@ -5,7 +5,7 @@
  *
  */
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <bitops.h>
 #include <memory.h>
 #include <boot_if.h>
diff --git a/lib/preboot/module_fwupd.c b/lib/preboot/module_fwupd.c
index 013cdf6..31b1e61 100644
--- a/lib/preboot/module_fwupd.c
+++ b/lib/preboot/module_fwupd.c
@@ -10,7 +10,7 @@
 #include <string.h>
 
 #include <boot_if.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <device.h>
 #include <hl_fw_if.h>
 #include <hl_fit_if.h>
diff --git a/lib/preboot/module_i2c_slave.c b/lib/preboot/module_i2c_slave.c
index aef0a3d..3782f36 100644
--- a/lib/preboot/module_i2c_slave.c
+++ b/lib/preboot/module_i2c_slave.c
@@ -6,7 +6,7 @@
  */
 
 #include <stdint.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <hl_i2c_slave_events.h>
 #include <hl_i2c_indirect.h>
 #include <boot_if.h>
diff --git a/lib/preboot/module_lkd.c b/lib/preboot/module_lkd.c
index 8a569ae..b411729 100644
--- a/lib/preboot/module_lkd.c
+++ b/lib/preboot/module_lkd.c
@@ -6,8 +6,8 @@
 
 #include <string.h>
 #include <hl_boot_if.h>
-#include <preboot.h>
-#include <hl_fw_if.h> /* requires preboot.h for __packed */
+#include <baremetal.h>
+#include <hl_fw_if.h> /* requires baremetal.h for __packed */
 #include <boot_if.h>
 #include <boot_info.h>
 #include <bitops.h>
diff --git a/lib/preboot/module_uart.c b/lib/preboot/module_uart.c
index 0695b20..aadb33b 100644
--- a/lib/preboot/module_uart.c
+++ b/lib/preboot/module_uart.c
@@ -4,7 +4,7 @@
  * All Rights Reserved.
  */
 
-#include <preboot.h>
+#include <baremetal.h>
 
 #include "cmd_module.h"
 #include "uart.h"
diff --git a/lib/preboot/thermal.c b/lib/preboot/thermal.c
index d7b9ab2..6296a8a 100644
--- a/lib/preboot/thermal.c
+++ b/lib/preboot/thermal.c
@@ -5,7 +5,7 @@
  *
  */
 #include <stdbool.h>
-#include <preboot.h>
+#include <baremetal.h>
 #include <pll.h>
 #include <efuse.h>
 #include <string.h>
diff --git a/lib/preboot/thermal.h b/lib/preboot/thermal.h
index c5a281c..169fa3a 100644
--- a/lib/preboot/thermal.h
+++ b/lib/preboot/thermal.h
@@ -8,7 +8,7 @@
 #ifndef _HL_THERMAL_H_
 #define _HL_THERMAL_H_
 
-#include <preboot.h>
+#include <baremetal.h>
 #include <stdbool.h>
 
 #define HL_ASIC_TS_MAX_COUNT    4
